lib/SILOptimizer/SILCombiner/SILCombinerApplyVisitors.cpp

```diff
@@ -1014,6 +1015,129 @@ class RAIIOpenedArchetypesTracker {
   }
 };
 
+/// Helper routine for propagating concrete type from
+/// class hierarchy analysis.
+bool SILCombiner::propagateConcreteTypeFromCTAInternal(
+    FullApplySite Apply, ProtocolDecl *Protocol, SILValue &Arg,
+    SILValue &NewArg, Optional<ProtocolConformanceRef> &ConformanceRef,
+    ArchetypeType *OpenedArchetype, CanType &ConcreteType) {
+
+  if (VisitedAIForCTA.find(Apply) != VisitedAIForCTA.end())
+    return false;
+  auto &M = Builder.getModule();
+  if (!Protocol)
```

Comment:
When is Protocol null? Either document it or remove this check if it can't happen